tests/integration/__tests__/early-exit.js

```diff
@@ -94,24 +94,22 @@ describe("throw error and show usage with something unexpected", () => {
 test("node version error", async () => {
   const originalProcessVersion = process.version;
 
-  try {
-    Object.defineProperty(process, "version", {
-      value: "v8.0.0",
-      writable: false,
-    });
-    const result = runPrettier("cli", ["--help"]);
-    expect(await result.status).toBe(1);
-    expect(await result.stderr).toBe(
-      `prettier requires at least version ${
-        isProduction ? "10.13.0" : "12.17.0"
-      } of Node, please upgrade\n`
-    );
-    expect(await result.stdout).toBe("");
-    expect(await result.write).toEqual([]);
-  } finally {
-    Object.defineProperty(process, "version", {
-      value: originalProcessVersion,
-      writable: false,
-    });
-  }
+  Object.defineProperty(process, "version", {
+    value: "v8.0.0",
+    writable: false,
+  });
+  const result = await runPrettier("cli", ["--help"]);
+  Object.defineProperty(process, "version", {
+    value: originalProcessVersion,
+    writable: false,
+  });
```

Comment:
This will not be called if `runPrettier` is rejected. So I think we should call `runPrettier` in `try` block and call this in `finally` block.

Reply:
If it's rejected, the test will fail, why we need that.

Reply:
Because changing `process.version` affects out of this test case.

If we add the test like below under the `test("node version error", async () => ...);`.

```js
test("checks version", () => {
  expect(process.version).toBe("v14.18.2");
});
```

if current node version is `14.18.2`, this test is failed via:

```
checks version

    expect(received).toBe(expected) // Object.is equality

    Expected: "v14.18.2"
    Received: "v8.0.0"

      116 |
      117 | test("checks version", () => {
    > 118 |   expect(process.version).toBe("v14.18.2");
          |                           ^
      119 | });
      120 |
```

Such tests are rarely written, but it's better to use `finally` just in case.

Reply:
You are right, I was not thinking right, I thought Jest will exit if one test fails.